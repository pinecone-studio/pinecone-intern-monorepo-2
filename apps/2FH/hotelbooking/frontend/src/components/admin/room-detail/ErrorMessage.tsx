import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { AlertCircle } from 'lucide-react';

interface ErrorMessageProps {
  message: string;
}

export const ErrorMessage = ({ message }: ErrorMessageProps) => {
  return (
    <div className="min-h-screen bg-gray-50 p-6">
      <div className="max-w-6xl mx-auto">
        <Card>
          <CardHeader>
            <CardTitle className="flex items-center gap-2 text-red-600">
              <AlertCircle className="h-5 w-5" />
              Error
            </CardTitle>
          </CardHeader>
          <CardContent>
            <p className="text-gray-700">{message}</p>
          </CardContent>
        </Card>
      </div>
    </div>
  );
};


