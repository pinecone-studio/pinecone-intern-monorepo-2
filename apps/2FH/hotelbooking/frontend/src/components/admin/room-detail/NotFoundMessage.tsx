/* eslint-disable  */
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { Search } from 'lucide-react';

export const NotFoundMessage = () => {
  return (
    <div className="min-h-screen bg-gray-50 p-6">
      <div className="max-w-6xl mx-auto">
        <Card>
          <CardHeader>
            <CardTitle className="flex items-center gap-2 text-gray-600">
              <Search className="h-5 w-5" />
              Room Not Found
            </CardTitle>
          </CardHeader>
          <CardContent>
            <p className="text-gray-700">The room you're looking for could not be found.</p>
          </CardContent>
        </Card>
      </div>
    </div>
  );
};
