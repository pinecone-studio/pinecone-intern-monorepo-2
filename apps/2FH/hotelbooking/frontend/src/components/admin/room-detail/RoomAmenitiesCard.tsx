'use client';
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { Button } from '@/components/ui/button';
import { Wifi, Utensils, Bed, Accessibility, Gamepad2, Coffee } from 'lucide-react';
import { EditRoomModal } from './EditRoomModal';

interface RoomAmenitiesCardProps {
  room: any;
  editModalState: {
    isOpen: boolean;
    section: 'basic' | 'amenities' | 'images' | 'details';
  };
  setEditModalState: (_state: { isOpen: boolean; section: 'basic' | 'amenities' | 'images' | 'details' }) => void;
  refetch: () => Promise<any>;
  roomId: string;
}

export const RoomAmenitiesCard = ({ room, editModalState, setEditModalState, refetch, roomId }: RoomAmenitiesCardProps) => {
  // Create a comprehensive list of all amenities with their values and labels
  const getAllAmenities = () => {
    const amenityCategories = [
      { key: 'internet', label: 'Internet', icon: <Wifi className="h-4 w-4" /> },
      { key: 'foodAndDrink', label: 'Food & Drink', icon: <Utensils className="h-4 w-4" /> },
      { key: 'bedRoom', label: 'Bedroom', icon: <Bed className="h-4 w-4" /> },
      { key: 'bathroom', label: 'Bathroom', icon: <Coffee className="h-4 w-4" /> },
      { key: 'accessibility', label: 'Accessibility', icon: <Accessibility className="h-4 w-4" /> },
      { key: 'entertainment', label: 'Entertainment', icon: <Gamepad2 className="h-4 w-4" /> },
      { key: 'other', label: 'Other', icon: <Coffee className="h-4 w-4" /> },
    ];

    const allAmenities: any[] = [];

    amenityCategories.forEach((category) => {
      const categoryValue = room[category.key];
      if (categoryValue) {
        // Handle both single values and arrays
        const values = Array.isArray(categoryValue) ? categoryValue : [categoryValue];
        values.forEach((value: string) => {
          if (value) {
            allAmenities.push({
              key: `${category.key}-${value}`,
              category: category.label,
              value: value,
              icon: category.icon,
            });
          }
        });
      }
    });

    return allAmenities;
  };

  const amenities = getAllAmenities();

  return (
    <Card>
      <CardHeader className="flex flex-row items-center justify-between">
        <CardTitle>Room Amenities</CardTitle>
        <EditRoomModal
          room={room}
          section="amenities"
          isOpen={editModalState.isOpen && editModalState.section === 'amenities'}
          onOpenChange={(open) => setEditModalState({ isOpen: open, section: 'amenities' })}
          refetch={refetch}
          roomId={roomId}
        />
        <Button size="sm" className="bg-blue-600 hover:bg-blue-700 text-white" onClick={() => setEditModalState({ isOpen: true, section: 'amenities' })}>
          Edit
        </Button>
      </CardHeader>
      <CardContent>
        {amenities.length === 0 ? (
          <div className="text-center py-8 text-gray-500">No amenities configured for this room</div>
        ) : (
          <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
            {amenities.map((amenity) => (
              <div key={amenity.key} className="flex items-center gap-3 p-3 border rounded-lg">
                <div className="text-blue-600">{amenity.icon}</div>
                <div className="flex-1">
                  <div className="font-medium text-gray-900">{amenity.value.replace(/_/g, ' ').replace(/\b\w/g, (l: string) => l.toUpperCase())}</div>
                  <div className="text-sm text-gray-600">{amenity.category}</div>
                </div>
              </div>
            ))}
          </div>
        )}
      </CardContent>
    </Card>
  );
};
