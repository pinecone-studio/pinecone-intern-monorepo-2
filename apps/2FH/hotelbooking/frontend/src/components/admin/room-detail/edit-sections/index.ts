export { BasicInfoSection } from './BasicInfoSection';
export { AmenitiesSection } from './AmenitiesSection';
export { ImagesSection } from './ImagesSection';
export { DetailsSection } from './DetailsSection';


