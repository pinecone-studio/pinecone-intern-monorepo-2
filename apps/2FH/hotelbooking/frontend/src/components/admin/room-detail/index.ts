export { RoomInfoCard } from './RoomInfoCard';
export { RoomAmenitiesCard } from './RoomAmenitiesCard';
export { RoomImagesCard } from './RoomImagesCard';
export { LoadingSkeleton } from './LoadingSkeleton';
export { ErrorMessage } from './ErrorMessage';
export { NotFoundMessage } from './NotFoundMessage';
export { RoomDetailPage } from './RoomDetailPage';
export { EditRoomModal } from './EditRoomModal';
