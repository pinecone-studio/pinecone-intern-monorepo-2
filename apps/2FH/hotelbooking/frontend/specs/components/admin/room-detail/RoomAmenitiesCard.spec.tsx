/* eslint-disable  */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { RoomAmenitiesCard } from '@/components/admin/room-detail/RoomAmenitiesCard';

// Mock dependencies
jest.mock('@/components/admin/room-detail/EditRoomModal', () => ({
  EditRoomModal: ({ room, section, isOpen, onOpenChange, refetch, roomId }: any) => (
    <div data-testid="edit-room-modal">
      <div>Edit Room Modal - {section}</div>
      <button onClick={() => onOpenChange(false)}>Close Modal</button>
    </div>
  ),
}));

describe('RoomAmenitiesCard', () => {
  const mockRoom = {
    id: 'room-1',
    name: 'Deluxe Suite',
    internet: ['free_wifi', 'free_wired_internet'],
    foodAndDrink: ['free_breakfast', 'minibar'],
    bedRoom: ['air_conditioner', 'bed_sheets'],
    bathroom: ['private', 'free_toiletries'],
    accessibility: ['wheelchair_accessible'],
    entertainment: ['tv', 'cable_channels'],
    other: ['desk', 'safe'],
  };

  const defaultProps = {
    room: mockRoom,
    editModalState: { isOpen: false, section: 'amenities' as const },
    setEditModalState: jest.fn(),
    refetch: jest.fn(),
    roomId: 'room-1',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders amenities card correctly', () => {
    render(<RoomAmenitiesCard {...defaultProps} />);

    expect(screen.getByText('Room Amenities')).toBeInTheDocument();
  });

  it('renders all amenity categories', () => {
    render(<RoomAmenitiesCard {...defaultProps} />);

    // Check for internet amenities
    expect(screen.getByText('Free Wifi')).toBeInTheDocument();
    expect(screen.getByText('Free Wired Internet')).toBeInTheDocument();
    const internetElements = screen.getAllByText('Internet');
    expect(internetElements.length).toBeGreaterThanOrEqual(2); // Should have at least 2 internet amenities

    // Check for food & drink amenities
    expect(screen.getByText('Free Breakfast')).toBeInTheDocument();
    expect(screen.getByText('Minibar')).toBeInTheDocument();
    const foodDrinkElements = screen.getAllByText('Food & Drink');
    expect(foodDrinkElements.length).toBeGreaterThanOrEqual(2); // Should have at least 2 food & drink amenities

    // Check for bedroom amenities
    expect(screen.getByText('Air Conditioner')).toBeInTheDocument();
    expect(screen.getByText('Bed Sheets')).toBeInTheDocument();
    const bedroomElements = screen.getAllByText('Bedroom');
    expect(bedroomElements.length).toBeGreaterThanOrEqual(2); // Should have at least 2 bedroom amenities

    // Check for bathroom amenities
    expect(screen.getByText('Private')).toBeInTheDocument();
    expect(screen.getByText('Free Toiletries')).toBeInTheDocument();
    const bathroomElements = screen.getAllByText('Bathroom');
    expect(bathroomElements.length).toBeGreaterThanOrEqual(2); // Should have at least 2 bathroom amenities

    // Check for accessibility amenities
    expect(screen.getByText('Wheelchair Accessible')).toBeInTheDocument();
    expect(screen.getByText('Accessibility')).toBeInTheDocument();

    // Check for entertainment amenities
    expect(screen.getByText('Tv')).toBeInTheDocument();
    expect(screen.getByText('Cable Channels')).toBeInTheDocument();
    const entertainmentElements = screen.getAllByText('Entertainment');
    expect(entertainmentElements.length).toBeGreaterThanOrEqual(2); // Should have at least 2 entertainment amenities

    // Check for other amenities
    expect(screen.getByText('Desk')).toBeInTheDocument();
    expect(screen.getByText('Safe')).toBeInTheDocument();
    // Note: "Other Amenities" category might not be rendered if there are no amenities in that category
  });

  it('handles single amenity values', () => {
    const roomWithSingleAmenities = {
      ...mockRoom,
      internet: 'free_wifi', // Single value instead of array
      foodAndDrink: 'free_breakfast',
    };

    render(<RoomAmenitiesCard {...defaultProps} room={roomWithSingleAmenities} />);

    expect(screen.getByText('Free Wifi')).toBeInTheDocument();
    expect(screen.getByText('Free Breakfast')).toBeInTheDocument();
  });

  it('handles empty amenities', () => {
    const roomWithNoAmenities = {
      ...mockRoom,
      internet: [],
      foodAndDrink: [],
      bedRoom: [],
      bathroom: [],
      accessibility: [],
      entertainment: [],
      other: [],
    };

    render(<RoomAmenitiesCard {...defaultProps} room={roomWithNoAmenities} />);

    expect(screen.getByText('No amenities configured for this room')).toBeInTheDocument();
  });

  it('handles null amenities', () => {
    const roomWithNullAmenities = {
      ...mockRoom,
      internet: null,
      foodAndDrink: null,
      bedRoom: null,
      bathroom: null,
      accessibility: null,
      entertainment: null,
      other: null,
    };

    render(<RoomAmenitiesCard {...defaultProps} room={roomWithNullAmenities} />);

    expect(screen.getByText('No amenities configured for this room')).toBeInTheDocument();
  });

  it('handles undefined amenities', () => {
    const roomWithUndefinedAmenities = {
      ...mockRoom,
      internet: undefined,
      foodAndDrink: undefined,
      bedRoom: undefined,
      bathroom: undefined,
      accessibility: undefined,
      entertainment: undefined,
      other: undefined,
    };

    render(<RoomAmenitiesCard {...defaultProps} room={roomWithUndefinedAmenities} />);

    expect(screen.getByText('No amenities configured for this room')).toBeInTheDocument();
  });

  it('filters out empty amenity values', () => {
    const roomWithEmptyValues = {
      ...mockRoom,
      internet: ['free_wifi', '', 'free_wired_internet'],
      foodAndDrink: [null, 'free_breakfast', undefined],
    };

    render(<RoomAmenitiesCard {...defaultProps} room={roomWithEmptyValues} />);

    expect(screen.getByText('Free Wifi')).toBeInTheDocument();
    expect(screen.getByText('Free Wired Internet')).toBeInTheDocument();
    expect(screen.getByText('Free Breakfast')).toBeInTheDocument();
  });

  it('formats amenity values correctly', () => {
    const roomWithUnderscores = {
      ...mockRoom,
      internet: ['free_wifi', 'wired_internet_connection'],
      foodAndDrink: ['free_breakfast_included'],
    };

    render(<RoomAmenitiesCard {...defaultProps} room={roomWithUnderscores} />);

    expect(screen.getByText('Free Wifi')).toBeInTheDocument();
    expect(screen.getByText('Wired Internet Connection')).toBeInTheDocument();
    expect(screen.getByText('Free Breakfast Included')).toBeInTheDocument();
  });

  it('renders amenities in grid layout', () => {
    render(<RoomAmenitiesCard {...defaultProps} />);

    const gridContainer = document.querySelector('.grid.grid-cols-1.md\\:grid-cols-2');
    expect(gridContainer).toBeInTheDocument();
  });

  it('renders amenity items with proper styling', () => {
    render(<RoomAmenitiesCard {...defaultProps} />);

    const amenityItems = document.querySelectorAll('.flex.items-center.gap-3.p-3.border.rounded-lg');
    expect(amenityItems.length).toBeGreaterThan(0);
  });

  it('opens edit modal when edit button is clicked', () => {
    const setEditModalState = jest.fn();
    render(<RoomAmenitiesCard {...defaultProps} setEditModalState={setEditModalState} />);

    const editButton = screen.getByText('Edit');
    fireEvent.click(editButton);

    expect(setEditModalState).toHaveBeenCalledWith({ isOpen: true, section: 'amenities' });
  });

  it('renders edit modal when open', () => {
    render(<RoomAmenitiesCard {...defaultProps} editModalState={{ isOpen: true, section: 'amenities' }} />);

    expect(screen.getByTestId('edit-room-modal')).toBeInTheDocument();
  });

  it('handles edit modal state changes', () => {
    const setEditModalState = jest.fn();
    render(<RoomAmenitiesCard {...defaultProps} setEditModalState={setEditModalState} editModalState={{ isOpen: true, section: 'amenities' }} />);

    // The modal should be open and the onOpenChange should be set up
    expect(screen.getByTestId('edit-room-modal')).toBeInTheDocument();

    // Simulate the onOpenChange callback being called
    const closeButton = screen.getByText('Close Modal');
    fireEvent.click(closeButton);

    // This should trigger the onOpenChange callback
    expect(screen.getByTestId('edit-room-modal')).toBeInTheDocument();
  });

  it('displays correct icons for amenity categories', () => {
    render(<RoomAmenitiesCard {...defaultProps} />);

    // Check that icons are rendered (they should be SVG elements)
    const icons = document.querySelectorAll('svg');
    expect(icons.length).toBeGreaterThan(0);
  });

  it('handles mixed amenity data types', () => {
    const roomWithMixedTypes = {
      ...mockRoom,
      internet: ['free_wifi'], // Array
      foodAndDrink: 'free_breakfast', // String
      bedRoom: null, // Null
      bathroom: undefined, // Undefined
    };

    render(<RoomAmenitiesCard {...defaultProps} room={roomWithMixedTypes} />);

    expect(screen.getByText('Free Wifi')).toBeInTheDocument();
    expect(screen.getByText('Free Breakfast')).toBeInTheDocument();
  });

  it('creates unique keys for amenities', () => {
    const roomWithDuplicateValues = {
      ...mockRoom,
      internet: ['free_wifi'],
      foodAndDrink: ['free_wifi'], // Same value in different category
    };

    render(<RoomAmenitiesCard {...defaultProps} room={roomWithDuplicateValues} />);

    // Should render both instances with different keys
    const freeWifiElements = screen.getAllByText('Free Wifi');
    expect(freeWifiElements.length).toBe(2);
  });

  it('matches snapshot', () => {
    const { container } = render(<RoomAmenitiesCard {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with no amenities', () => {
    const roomWithNoAmenities = {
      ...mockRoom,
      internet: [],
      foodAndDrink: [],
      bedRoom: [],
      bathroom: [],
      accessibility: [],
      entertainment: [],
      other: [],
    };

    const { container } = render(<RoomAmenitiesCard {...defaultProps} room={roomWithNoAmenities} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
