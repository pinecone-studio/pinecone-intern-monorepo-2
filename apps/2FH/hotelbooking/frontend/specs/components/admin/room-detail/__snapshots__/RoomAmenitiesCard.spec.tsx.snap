// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomAmenitiesCard matches snapshot 1`] = `
<div
  class="rounded-lg border bg-card text-card-foreground shadow-sm"
>
  <div
    class="space-y-1.5 p-6 flex flex-row items-center justify-between"
  >
    <h3
      class="text-2xl font-semibold leading-none tracking-tight"
    >
      Room Amenities
    </h3>
    <div
      data-testid="edit-room-modal"
    >
      <div>
        Edit Room Modal - 
        amenities
      </div>
      <button>
        Close Modal
      </button>
    </div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 h-9 rounded-md px-3 bg-blue-600 hover:bg-blue-700 text-white"
    >
      Edit
    </button>
  </div>
  <div
    class="p-6 pt-0"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-2 gap-4"
    >
      <div
        class="flex items-center gap-3 p-3 border rounded-lg"
      >
        <div
          class="text-blue-600"
        >
          <svg
            class="lucide lucide-wifi h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 20h.01"
            />
            <path
              d="M2 8.82a15 15 0 0 1 20 0"
            />
            <path
              d="M5 12.859a10 10 0 0 1 14 0"
            />
            <path
              d="M8.5 16.429a5 5 0 0 1 7 0"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="font-medium text-gray-900"
          >
            Free Wifi
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Internet
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-3 border rounded-lg"
      >
        <div
          class="text-blue-600"
        >
          <svg
            class="lucide lucide-wifi h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 20h.01"
            />
            <path
              d="M2 8.82a15 15 0 0 1 20 0"
            />
            <path
              d="M5 12.859a10 10 0 0 1 14 0"
            />
            <path
              d="M8.5 16.429a5 5 0 0 1 7 0"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="font-medium text-gray-900"
          >
            Free Wired Internet
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Internet
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-3 border rounded-lg"
      >
        <div
          class="text-blue-600"
        >
          <svg
            class="lucide lucide-utensils h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 2v7c0 1.1.9 2 2 2h4a2 2 0 0 0 2-2V2"
            />
            <path
              d="M7 2v20"
            />
            <path
              d="M21 15V2a5 5 0 0 0-5 5v6c0 1.1.9 2 2 2h3Zm0 0v7"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="font-medium text-gray-900"
          >
            Free Breakfast
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Food & Drink
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-3 border rounded-lg"
      >
        <div
          class="text-blue-600"
        >
          <svg
            class="lucide lucide-utensils h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 2v7c0 1.1.9 2 2 2h4a2 2 0 0 0 2-2V2"
            />
            <path
              d="M7 2v20"
            />
            <path
              d="M21 15V2a5 5 0 0 0-5 5v6c0 1.1.9 2 2 2h3Zm0 0v7"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="font-medium text-gray-900"
          >
            Minibar
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Food & Drink
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-3 border rounded-lg"
      >
        <div
          class="text-blue-600"
        >
          <svg
            class="lucide lucide-bed h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 4v16"
            />
            <path
              d="M2 8h18a2 2 0 0 1 2 2v10"
            />
            <path
              d="M2 17h20"
            />
            <path
              d="M6 8v9"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="font-medium text-gray-900"
          >
            Air Conditioner
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Bedroom
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-3 border rounded-lg"
      >
        <div
          class="text-blue-600"
        >
          <svg
            class="lucide lucide-bed h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 4v16"
            />
            <path
              d="M2 8h18a2 2 0 0 1 2 2v10"
            />
            <path
              d="M2 17h20"
            />
            <path
              d="M6 8v9"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="font-medium text-gray-900"
          >
            Bed Sheets
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Bedroom
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-3 border rounded-lg"
      >
        <div
          class="text-blue-600"
        >
          <svg
            class="lucide lucide-coffee h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 2v2"
            />
            <path
              d="M14 2v2"
            />
            <path
              d="M16 8a1 1 0 0 1 1 1v8a4 4 0 0 1-4 4H7a4 4 0 0 1-4-4V9a1 1 0 0 1 1-1h14a4 4 0 1 1 0 8h-1"
            />
            <path
              d="M6 2v2"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="font-medium text-gray-900"
          >
            Private
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Bathroom
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-3 border rounded-lg"
      >
        <div
          class="text-blue-600"
        >
          <svg
            class="lucide lucide-coffee h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 2v2"
            />
            <path
              d="M14 2v2"
            />
            <path
              d="M16 8a1 1 0 0 1 1 1v8a4 4 0 0 1-4 4H7a4 4 0 0 1-4-4V9a1 1 0 0 1 1-1h14a4 4 0 1 1 0 8h-1"
            />
            <path
              d="M6 2v2"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="font-medium text-gray-900"
          >
            Free Toiletries
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Bathroom
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-3 border rounded-lg"
      >
        <div
          class="text-blue-600"
        >
          <svg
            class="lucide lucide-accessibility h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="16"
              cy="4"
              r="1"
            />
            <path
              d="m18 19 1-7-6 1"
            />
            <path
              d="m5 8 3-3 5.5 3-2.36 3.5"
            />
            <path
              d="M4.24 14.5a5 5 0 0 0 6.88 6"
            />
            <path
              d="M13.76 17.5a5 5 0 0 0-6.88-6"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="font-medium text-gray-900"
          >
            Wheelchair Accessible
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Accessibility
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-3 border rounded-lg"
      >
        <div
          class="text-blue-600"
        >
          <svg
            class="lucide lucide-gamepad2 h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="6"
              x2="10"
              y1="11"
              y2="11"
            />
            <line
              x1="8"
              x2="8"
              y1="9"
              y2="13"
            />
            <line
              x1="15"
              x2="15.01"
              y1="12"
              y2="12"
            />
            <line
              x1="18"
              x2="18.01"
              y1="10"
              y2="10"
            />
            <path
              d="M17.32 5H6.68a4 4 0 0 0-3.978 3.59c-.006.052-.01.101-.017.152C2.604 9.416 2 14.456 2 16a3 3 0 0 0 3 3c1 0 1.5-.5 2-1l1.414-1.414A2 2 0 0 1 9.828 16h4.344a2 2 0 0 1 1.414.586L17 18c.5.5 1 1 2 1a3 3 0 0 0 3-3c0-1.545-.604-6.584-.685-7.258-.007-.05-.011-.1-.017-.151A4 4 0 0 0 17.32 5z"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="font-medium text-gray-900"
          >
            Tv
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Entertainment
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-3 border rounded-lg"
      >
        <div
          class="text-blue-600"
        >
          <svg
            class="lucide lucide-gamepad2 h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="6"
              x2="10"
              y1="11"
              y2="11"
            />
            <line
              x1="8"
              x2="8"
              y1="9"
              y2="13"
            />
            <line
              x1="15"
              x2="15.01"
              y1="12"
              y2="12"
            />
            <line
              x1="18"
              x2="18.01"
              y1="10"
              y2="10"
            />
            <path
              d="M17.32 5H6.68a4 4 0 0 0-3.978 3.59c-.006.052-.01.101-.017.152C2.604 9.416 2 14.456 2 16a3 3 0 0 0 3 3c1 0 1.5-.5 2-1l1.414-1.414A2 2 0 0 1 9.828 16h4.344a2 2 0 0 1 1.414.586L17 18c.5.5 1 1 2 1a3 3 0 0 0 3-3c0-1.545-.604-6.584-.685-7.258-.007-.05-.011-.1-.017-.151A4 4 0 0 0 17.32 5z"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="font-medium text-gray-900"
          >
            Cable Channels
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Entertainment
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-3 border rounded-lg"
      >
        <div
          class="text-blue-600"
        >
          <svg
            class="lucide lucide-coffee h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 2v2"
            />
            <path
              d="M14 2v2"
            />
            <path
              d="M16 8a1 1 0 0 1 1 1v8a4 4 0 0 1-4 4H7a4 4 0 0 1-4-4V9a1 1 0 0 1 1-1h14a4 4 0 1 1 0 8h-1"
            />
            <path
              d="M6 2v2"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="font-medium text-gray-900"
          >
            Desk
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Other
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-3 p-3 border rounded-lg"
      >
        <div
          class="text-blue-600"
        >
          <svg
            class="lucide lucide-coffee h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 2v2"
            />
            <path
              d="M14 2v2"
            />
            <path
              d="M16 8a1 1 0 0 1 1 1v8a4 4 0 0 1-4 4H7a4 4 0 0 1-4-4V9a1 1 0 0 1 1-1h14a4 4 0 1 1 0 8h-1"
            />
            <path
              d="M6 2v2"
            />
          </svg>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="font-medium text-gray-900"
          >
            Safe
          </div>
          <div
            class="text-sm text-gray-600"
          >
            Other
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RoomAmenitiesCard matches snapshot with no amenities 1`] = `
<div
  class="rounded-lg border bg-card text-card-foreground shadow-sm"
>
  <div
    class="space-y-1.5 p-6 flex flex-row items-center justify-between"
  >
    <h3
      class="text-2xl font-semibold leading-none tracking-tight"
    >
      Room Amenities
    </h3>
    <div
      data-testid="edit-room-modal"
    >
      <div>
        Edit Room Modal - 
        amenities
      </div>
      <button>
        Close Modal
      </button>
    </div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 h-9 rounded-md px-3 bg-blue-600 hover:bg-blue-700 text-white"
    >
      Edit
    </button>
  </div>
  <div
    class="p-6 pt-0"
  >
    <div
      class="text-center py-8 text-gray-500"
    >
      No amenities configured for this room
    </div>
  </div>
</div>
`;
